boringssl: Add support for RISC-V 64-bit architecture

SpriteOvO opened the pull request:
Hi! I'm building this package for ArchLinux RISC-V distribution and I get an error message:

```
-- Checking for module 'libunwind-generic'
--   Package 'libunwind-generic', required by 'virtual:world', not found
libunwind not found. Disabling unwind tests.
CMake Error at vendor/boringssl/CMakeLists.txt:502 (message):
  Unknown processor:riscv64
```

`boringssl` has supported RISC-V 64 architecture in [this commit](https://boringssl.googlesource.com/boringssl/+/4566bb5fe517f7f141b5fe935c559fc4311af35d). This PR brings it to the patch set of this package to support compiling it for RISC-V 64 architecture.

And since `git am` doesn't support fuzzy patching, this patch cannot be applied by `git am` directly, I replaced it with `patch` command, it shouldn't ruin the previous work, and avoid depending on `git` for this purpose.

sgn commented:
I don't think we need fuzzy patching. Patch should be applied and rebased for each release. And patching, thus git is only need to build from git repository (which means git is required), when tarball is generated, all patches are applied.
IOW, I don't think second patch is welcomed, of course @nmeum  has the final words.

SpriteOvO commented:
> And patching, thus git is only need to build from git repository (which means git is required)

Just saying, if someone wants to build from the git repo in a clean `chroot` environment (whatever manually or from a build script), `git am` will ask

```
*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address
```

which doesn't make sense for the purpose of applying patches. It's not a big deal though.

sgn commented:
> > And patching, thus git is only need to build from git repository (which means git is required)
> 
> Just saying, if someone wants to build from the git repo in a clean `chroot` environment (whatever manually or from a build script), `git am` will ask
> 
> ```
> *** Please tell me who you are.
> 
> Run
> 
>   git config --global user.email "you@example.com"
>   git config --global user.name "Your Name"
> 
> to set your account's default identity.
> Omit --global to set the identity only in this repository.
> 
> fatal: unable to auto-detect email address
> ```
> 
> which doesn't make sense for the purpose of applying patches. It's not a big deal though.

That someone could just (`.git` could be change to `$(git rev-parse --git-dir)`, but whatever):
```sh
$ git config user.name "some one" && git config user.email "mail@example.com"
$ # or
$ cat <<EOF >.git/config
[user]
name = Some One
email = mail@example.com
EOF
```

nmeum commented:
Can you rebase this against master?

I think I would also prefer just merging the RISC-V patch for now and discussing the `git am` change later separately.

SpriteOvO commented:
> I think I would also prefer just merging the RISC-V patch for now and discussing the `git am` change later separately.

@nmeum I have removed the changes to `git am` and updated the patch to avoid fuzzy patching.

nmeum commented:
> > I think I would also prefer just merging the RISC-V patch for now and discussing the `git am` change later separately.
> 
> @nmeum I have removed the changes to `git am` and updated the patch to avoid fuzzy patching.

Thanks! Merged your changes :tada:
